nifi-extension-bundles/nifi-extension-utils/nifi-file-transfer/src/main/java/org/apache/nifi/processor/util/file/transfer/FetchFileTransfer.java

```diff
@@ -363,7 +379,45 @@ private void performCompletionStrategy(final FileTransfer transfer, final Proces
                     transfer.ensureDirectoryExists(flowFile, new File(absoluteTargetDirPath));
                 }
 
-                final String destinationPath = String.format("%s/%s", absoluteTargetDirPath, simpleFilename);
+                String destinationFileName = simpleFilename;
+                final FileInfo remoteFileInfo = transfer.getRemoteFileInfo(flowFile, absoluteTargetDirPath, destinationFileName);
+                if (remoteFileInfo != null) {
+                    final String strategy = context.getProperty(MOVE_CONFLICT_RESOLUTION).getValue();
+                    getLogger().info("Detected filename conflict moving remote file for {} so handling using configured Conflict Resolution of {}", flowFile, strategy);
+                    switch (strategy.toUpperCase()) {
+                        case FileTransfer.CONFLICT_RESOLUTION_REPLACE:
+                            try {
+                                transfer.deleteFile(flowFile, absoluteTargetDirPath, destinationFileName);
+                            } catch (final IOException deleteEx) {
+                                getLogger().warn("Failed to delete existing destination file {} on {}:{} due to {}. Move will be attempted regardless.",
+                                        destinationFileName, host, port, deleteEx.toString(), deleteEx);
```

Comment:
The `due to` syntax should be avoided, the inclusion of the exception provides both the message and the stack trace.
```suggestion
                                getLogger().warn("Failed to delete existing destination file {} on {}:{}. Move will be attempted regardless.",
                                        destinationFileName, host, port, deleteEx);
```